### ISSUE TITLE
Record collection can pass with wrong solution

### ISSUE THREAD

**Issue by Ieahleen** [freeCodeCamp/freeCodeCamp#38962]
**Describe your problem and how to reproduce it:**
This challenge has this test that requires the creation of the property: `After updateRecords(5439, "tracks", "Take a Chance on Me"), tracks should have "Take a Chance on Me" as the last element.`
If the code contain something like this:
```
} else if (prop == "tracks" && collection[id].hasOwnProperty("tracks") == false) {
  collection[id][prop] = [];
} else if (prop == "tracks" && value !== "") {
  collection[id].tracks.push(value);
}
```
in which the creation of the array and the push method are used in different if statements, it should not pass, but if for debugging someone adds the function call `updateRecords(5439, "tracks", "Take a Chance on Me")`, then the tests pass.

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/record-collection

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
![image](https://user-images.githubusercontent.com/26656284/83327182-4d036780-a27a-11ea-858f-c65c22e1ce98.png)

**Comment by Sky020** [comment]
The only way I can think of testing around this is to parse the code in such a way so as to remove all calls to the function, before tests.

**Comment by ojeytonwilliams** [comment]
How about resetting `collection` to the expected object in `js-teardown`?  That way the user can `updateRecords` however they like, but we always test the same data.

**Comment by niteshseram** [comment]
This code is passing the test because `updateRecords()` function is called twice. One in the code section and another while doing assert. Just removing the `updateRecords()` from the code section might solved the problem if I am wrong.

**Comment by Sky020** [comment]
Let us just fix this and: #38691

**Comment by niteshseram** [comment]
I was trying to fix but even I faced this #38691 issue. I couldn't figured out why?
